src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfoCacheService.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Collections;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.FindSymbols.SymbolTree
+{
+    [ExportWorkspaceService(typeof(SymbolTreeInfoCacheService)), Shared]
+    internal sealed partial class SymbolTreeInfoCacheService : IWorkspaceService
```

Comment:
this code moved nearly verbatim from https://github.com/dotnet/roslyn/pull/63438/files#diff-31deb8089909b5074d41add123e9ce6cc85ad536dd6c4030b552cd4f55ee73a2

Reply:
(i really didn't change anything here.  feel free to review, but it's just following the logic from before).